exchange/auction.go

```diff
@@ -107,29 +108,36 @@ func (d *DebugLog) PutDebugLogError(cache prebid_cache_client.Client, timeout in
 
 func newAuction(seatBids map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, numImps int, preferDeals bool) *auction {
 	winningBids := make(map[string]*entities.PbsOrtbBid, numImps)
-	winningBidsByBidder := make(map[string]map[openrtb_ext.BidderName]*entities.PbsOrtbBid, numImps)
+	winningBidsByBidder := make(map[string]map[openrtb_ext.BidderName][]*entities.PbsOrtbBid, numImps)
 
 	for bidderName, seatBid := range seatBids {
 		if seatBid != nil {
 			for _, bid := range seatBid.Bids {
-				cpm := bid.Bid.Price
 				wbid, ok := winningBids[bid.Bid.ImpID]
 				if !ok || isNewWinningBid(bid.Bid, wbid.Bid, preferDeals) {
 					winningBids[bid.Bid.ImpID] = bid
 				}
+
 				if bidMap, ok := winningBidsByBidder[bid.Bid.ImpID]; ok {
-					bestSoFar, ok := bidMap[bidderName]
-					if !ok || cpm > bestSoFar.Bid.Price {
-						bidMap[bidderName] = bid
-					}
+					bidMap[bidderName] = append(bidMap[bidderName], bid)
```

Comment:
For multi-bid targeting, we need to process bids per adapter grouped by `ImpID`, hence used this original code to avoid rebuilding the same map in targeting.go's `setTargeting()`